implementations/rust/ockam/ockam_identity/src/trust_context.rs

```diff
@@ -0,0 +1,119 @@
+use async_trait::async_trait;
+
+use crate::{credential::Credential, error::IdentityError, Identity, PublicIdentity};
+use ockam_core::compat::{boxed::Box, string::String, sync::Arc};
+
+/// Trust Context is a set of information about a trusted authority
+#[derive(Clone)]
+pub struct TrustContext {
+    id: String,
```

Comment:
It seems to me that we should be more specific here and write `id: IdentityIdentifier` or `id: PublicIdentity`. Because we are not using arbitrary strings to create a trust context, it looks like we are always using a specific identity. I think that this is important when we try to understand what exact ABAC rule applies when we then use a `subject.trust_context_id` and `resource.trusted_context_id`. Maybe if the `trusted_context_id` field was named `trusted_authority_id` or something like that, it would make the data provenance clearer.

Reply:
I would love a stricter definition on this field;

However, it's also meant to be a generic field. For example we use the project id, in the use case of an orchestrator based trust context.

In the cases when that isn't available, I've been using IdentityIdentifier as a string, however, it could also just as simple be a uuid/guid.

**The only usage of it right now is for ABAC rules.**

Reply:
Well it looks like I was wrong anyway to suggest this if a trust context can eventually encompass several authorities. But then the semantics of the ABAC rules are less clear to me. If a server has a resource with attribute trust_context_1 connected to authority 1 and 2, can a client still access the resource if it has a trust_context only connected to authority 2?

Reply:
> The semantics of the ABAC rules are less clear to me. If a server has a resource with attribute trust_context_1 connected to authority 1 and 2, can a client still access the resource if it has a trust_context only connected to authority 2?

That is a really good point, I would think the ID should be tied to the authority specifically, which changes some things.

I'll write a note/todo about that, in the structs so when we get to that point we will know to change it.